src/core/node.ts

```diff
@@ -103,7 +112,7 @@ export abstract class Node /* TODO: implements INode*/ {
     // TODO: needs improvements, now called too often. Should be set propertly when applying snapshots / calling factories immediately!
     // TODO: should not be possible to change just subpath?
     setParent(newParent: Node | null, subpath: string | null = null) {
-        if (this.parent === newParent && this.pathParts[this.pathParts.length - 1] === subpath)
```

Comment:
Ideally, that subpath was used before only to update the children paths. This if avoided unecessary subpath updates. Since now it is computed, we could safely set the parent whenever we want and the path will be computed accordly.